### PR TITLE
net-mail/vacation: upgrade to EAPI7

### PR DESCRIPTION
**Pull request by usergeorge**
fixes build error on incorrect man page path

Bug: https://bugs.gentoo.org/696252
Signed-off-by: Jörg Habenicht <j.habenicht@gmx.de>
Package-Manager: Portage-2.3.76, Repoman-2.3.16

### CODE REVIEW

File: net-mail/vacation/vacation-1.2.7.0-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+DEPEND="${RDEPEND}
+	!mail-mta/sendmail"
+
+src_unpack() {
+	unpack ${A}
+	cd "${S}"
+}
+
+src_prepare() {
+	eapply_user
```
Comment (ConiKost) [inline net-mail/vacation/vacation-1.2.7.0-r1.ebuild]: Better call ```default```, as this on also calls later ```eapply_user```.

File: net-mail/vacation/vacation-1.2.7.0-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+	sed -i -e "s:install -s -m:install -m:" Makefile || die "sed failed!"
+	sed -i -e "s:-Xlinker:${LDFLAGS} -Xlinker:" Makefile || die "sed failed!"
+}
+
+src_compile () {
+	emake CC=$(tc-getCC) ARCH=$(tc-arch-kernel) CFLAGS="${CFLAGS} -DMAIN" || die "emake failed."
```
Comment (ConiKost) [inline net-mail/vacation/vacation-1.2.7.0-r1.ebuild]: emake has it's own internal die, no need here.

File: net-mail/vacation/vacation-1.2.7.0-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+LICENSE="GPL-2"
+KEYWORDS="~alpha ~amd64 ~x86"
+SLOT="0"
+
+RDEPEND="virtual/mta
+	sys-libs/gdbm"
+DEPEND="${RDEPEND}
+	!mail-mta/sendmail"
+
+src_unpack() {
```
Comment (ConiKost) [inline net-mail/vacation/vacation-1.2.7.0-r1.ebuild]: What's the purpose of unpack and then ```cd```? Can be maybe the entire section be dropped?

File: net-mail/vacation/vacation-1.2.7.0-r1.ebuild (hunk 2 above)
Comment (usergeorge) [reply net-mail/vacation/vacation-1.2.7.0-r1.ebuild]: Ok, will do.
Thank you

File: net-mail/vacation/vacation-1.2.7.0-r1.ebuild (hunk 1 above)
Comment (usergeorge) [reply net-mail/vacation/vacation-1.2.7.0-r1.ebuild]: Ok, will do.
Thank you

File: net-mail/vacation/vacation-1.2.7.0-r1.ebuild (hunk 3 above)
Comment (usergeorge) [reply net-mail/vacation/vacation-1.2.7.0-r1.ebuild]: Maybe you're right.
I just took over and did not validate the code.
I'll just remove it now

File: net-mail/vacation/vacation-1.2.7.0-r1.ebuild [hunk 4]
```diff
@@ -17,25 +17,19 @@ RDEPEND="virtual/mta
...
 	sed -i -e "s:-Xlinker:${LDFLAGS} -Xlinker:" Makefile || die "sed failed!"
 }
 
 src_compile () {
-	emake CC=$(tc-getCC) ARCH=$(tc-arch-kernel) CFLAGS="${CFLAGS} -DMAIN" || die "emake failed."
+	emake CC=$(tc-getCC) ARCH=$(tc-arch-kernel) CFLAGS="${CFLAGS} -DMAIN"
 }
 
 src_install () {
 	dodir /usr/bin
```
Comment (Zlogene) [inline net-mail/vacation/vacation-1.2.7.0-r1.ebuild]: why do you need this?

File: net-mail/vacation/vacation-1.2.7.0-r1.ebuild [hunk 5]
```diff
@@ -17,25 +17,19 @@ RDEPEND="virtual/mta
...
 	dodir /usr/share/man/man1
```
Comment (Zlogene) [inline net-mail/vacation/vacation-1.2.7.0-r1.ebuild]: and this?

File: net-mail/vacation/vacation-1.2.7.0-r1.ebuild [hunk 6]
```diff
@@ -17,25 +17,19 @@ RDEPEND="virtual/mta
...
-src_unpack() {
-	unpack ${A}
-	cd "${S}"
-}
-
 src_prepare() {
-	eapply_user
+	default
 
 	sed -i -e "s:install -s -m:install -m:" Makefile || die "sed failed!"
```
Comment (Zlogene) [inline net-mail/vacation/vacation-1.2.7.0-r1.ebuild]: why not multiple regexs sed?

File: net-mail/vacation/vacation-1.2.7.0-r1.ebuild (hunk 4 above)
Comment (usergeorge) [reply net-mail/vacation/vacation-1.2.7.0-r1.ebuild]: by request from ConiKost
https://github.com/gentoo/gentoo/pull/13499#discussion_r340735966

Comment (usergeorge) [reply net-mail/vacation/vacation-1.2.7.0-r1.ebuild]: Well I just took over editing this ebuild (to keep it in the ebuild db).
I don't know the exact reason why the additional variables are needed.
But it works well at my place

File: net-mail/vacation/vacation-1.2.7.0-r1.ebuild (hunk 6 above)
Comment (usergeorge) [reply net-mail/vacation/vacation-1.2.7.0-r1.ebuild]: Hmm, if I understand correctly you suggest to merge 2 sed calls into one?
Wouldn't do this, the execution time won't drop and the maintenance gets harder with 1 sed call

File: net-mail/vacation/vacation-1.2.7.0-r1.ebuild (hunk 4 above)
Comment (usergeorge) [reply net-mail/vacation/vacation-1.2.7.0-r1.ebuild]: I just gave it a try, CC and ARCH are not needed.
But the MAIN definition is needed, there are references in rfc822.c

I corrected the make call

File: net-mail/vacation/vacation-1.2.7.0-r1.ebuild (hunk 6 above)
Comment (usergeorge) [reply net-mail/vacation/vacation-1.2.7.0-r1.ebuild]: Or maybe I misunderstood you. Do you got a code example?

File: net-mail/vacation/vacation-1.2.7.0-r1.ebuild [hunk 7]
```diff
@@ -0,0 +1,35 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit eutils toolchain-funcs
```
Comment (juippis) [inline net-mail/vacation/vacation-1.2.7.0-r1.ebuild]: Hmm, I didn't see eutils being used here.

File: net-mail/vacation/vacation-1.2.7.0-r1.ebuild [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+DESCRIPTION="automatic mail answering program"
+HOMEPAGE="http://vacation.sourceforge.net/"
+SRC_URI="mirror://sourceforge/vacation/${P}.tar.gz"
+LICENSE="GPL-2"
+KEYWORDS="~alpha ~amd64 ~x86"
+SLOT="0"
+
+RDEPEND="virtual/mta
```
Comment (juippis) [inline net-mail/vacation/vacation-1.2.7.0-r1.ebuild]: Please sort by putting this below sys-libs.

File: net-mail/vacation/vacation-1.2.7.0-r1.ebuild [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+	sys-libs/gdbm"
+DEPEND="${RDEPEND}
+	!mail-mta/sendmail"
```
Comment (juippis) [inline net-mail/vacation/vacation-1.2.7.0-r1.ebuild]: Soft blockers shouldn't be DEPEND-only, as package manager can't handle that smartly on its own. Either add a hard blocker here, or move this to RDEPEND (so flip DEPEND and RDEPEND around basically).

File: net-mail/vacation/vacation-1.2.7.0-r1.ebuild (hunk 6 above)
Comment (juippis) [reply net-mail/vacation/vacation-1.2.7.0-r1.ebuild]: Of course it does, you don't have to open and close the same file twice. Less I/O, memory etc.

Besides it's easier to see what is being done to a single file, less chance of failures...

**Pull request closed without merging**